packages/@uppy/transloadit/src/index.ts

```diff
@@ -876,85 +790,39 @@ export default class Transloadit<
     return assembly
   }
 
-  #prepareUpload = async (fileIDs: string[], uploadID: string) => {
-    const files = fileIDs.map((id) => this.uppy.getFile(id))
-    const filesWithoutErrors = files.filter((file) => {
-      if (!file.error) {
-        this.uppy.emit('preprocess-progress', file, {
-          mode: 'indeterminate',
-          message: this.i18n('creatingAssembly'),
-        })
-        return true
-      }
-      return false
-    })
+  #prepareUpload = async (fileIDs: string[]) => {
+    const assemblyOptions = (
+      typeof this.opts.assemblyOptions === 'function' ?
+        await this.opts.assemblyOptions()
+      : this.opts.assemblyOptions) as OptionsWithRestructuredFields
```

Comment:
shouldn't we provide the files for the user in the `assemblyOptions()` function? I'd imagine that would be very useful. (like before)

```suggestion
    let assemblyOptions: OptionsWithRestructuredFields

    if (typeof this.opts.assemblyOptions === 'function') {
      const files = fileIDs.map((id) => this.uppy.getFile(id))
      assemblyOptions = await this.opts.assemblyOptions(files)
    } else {
      assemblyOptions = this.opts.assemblyOptions
    }
```

Reply:
Why would that be useful? This call is used to get your auth key and secret and your params. Besides, you can get anything you want inside there. Most likely, you will want global metadata (`uppy.getMeta()`) but if you want all files, you just call `uppy.getFiles()`. It's all one call away, and we keep this simple and fast.

Reply:
Yea i agree. I was just puzzled why the argument was there before but now removed